toolchain/driver/compile_subcommand.cpp

```diff
@@ -294,6 +295,35 @@ Whether to emit DWARF debug information.
         arg_b.Default(true);
         arg_b.Set(&include_debug_info);
       });
+  b.AddFlag(
+      {
+          .name = "poison_verbose",
```

Comment:
We've been using hyphens in flags (see line 289 above, etc)
```suggestion
          .name = "poison-verbose",
```

Reply:
Yeah, this makes the story for test_file passing through the flag significantly more complicated, see the end of the PR description for a brief explanation, but I can expand on it if you'd like. We would have to implement our own command line parsing to support this AFAICT.

Reply:
Can you elaborate? I think consistency in the way we name flags in the command line is pretty important, more than test complexity.

Note, another option would be to just have `file_test` call the `SetPoison` functions directly, instead of using the driver as an intermediary -- that'd also remove a lot of the threading concerns I've noted. But I can see you might want the ability to run the command line directly.

Reply:
Yes, absolutely.

The first issue is in absl flag parsing, which requires flags to use underscore separators and not hyphens, since the macros use the flag name as a c++ identifier.

It is possible to have absl parse flags it knows about and tell you the flags it didn't, with `absl::ParseAbseilFlagsOnly`. So then that would give us back `--poison-verbose` and `--poison-abort` which sounds really promising. But the return type there is a vector of `absl::UnrecognizedFlag`, and in that structure the `flag_name` field only contains what it says: the name, not the value. So `--poison-abort=impl:911` only gives us back `--poison-abort`. We lost the stuff after the `=`.

This means that to forward these flags we need to look for them in argv before `absl::ParseCommandLine()`, and remove them from the array. This is what I mean by writing our own command line parsing (of course, not the full complexity of ParseCommandLine()).

I am fine to do that if you prefer, I thought it would be very disliked, but I also dislike mixing the `-` and `_` in the toolchain flags.

The challenge would return for knowing which flags to pull out in Main() though. We don't have a FileTestBase to query from. So it may need to become global knowledge of //testing again?